src/arraydist.jl

```diff
@@ -18,9 +18,7 @@ function arraydist(dists::AbstractMatrix{<:UnivariateDistribution})
     return MatrixOfUnivariate(dists)
 end
 function Distributions._logpdf(dist::MatrixOfUnivariate, x::AbstractMatrix{<:Real})
-    # return sum(((d, xi),) -> logpdf(d, xi), zip(dist.dists, x))
-    # Broadcasting here breaks Tracker for some reason
-    return sum(map(logpdf, dist.dists, x))
+    return mapreduce(logpdf, +, dist.dists, x)
```

Comment:
> Are you sure this is helpful? At least it should still allocate since it just calls `reduce(+, map(logpdf, dist.dists, x))` So it seems to be the same as before but less obvious?

It guess you are right. Well, this is quite unfortunate, since the doc of `mapreduce()` explicitly states that it is there to avoid allocation.
Calling low-level broadcast routines each time you want `mapreduce()` looks like a design deficiency on the core Julia side.
There is JuliaLang/julia#41001 that should address it.
But meanwhile I'll just apply your suggestion.